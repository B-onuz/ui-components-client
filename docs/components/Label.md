# Label (Component)

Componente de label para campos de text, select, checkbox etc.

## Como utilizar

```js
import Label from 'ui-components/lib/components/Label'
```

```jsx
<Label>Foo Bar</Label>
```
