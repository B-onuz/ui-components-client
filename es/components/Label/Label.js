"use strict";

Object.defineProperty(exports, "__esModule", {
  value: true
});
exports["default"] = void 0;

var _react = _interopRequireDefault(require("react"));

var _styledComponents = _interopRequireDefault(require("styled-components"));

var _styledSystem = require("styled-system");

var _this = void 0,
    _jsxFileName = "/Users/master/projects/bonuz/ui-components/src/components/Label/Label.js";

function _interopRequireDefault(obj) { return obj && obj.__esModule ? obj : { "default": obj }; }

function _extends() { _extends = Object.assign || function (target) { for (var i = 1; i < arguments.length; i++) { var source = arguments[i]; for (var key in source) { if (Object.prototype.hasOwnProperty.call(source, key)) { target[key] = source[key]; } } } return target; }; return _extends.apply(this, arguments); }

function _objectWithoutPropertiesLoose(source, excluded) { if (source == null) return {}; var target = {}; var sourceKeys = Object.keys(source); var key, i; for (i = 0; i < sourceKeys.length; i++) { key = sourceKeys[i]; if (excluded.indexOf(key) >= 0) continue; target[key] = source[key]; } return target; }

function _templateObject() {
  var data = _taggedTemplateLiteralLoose(["\n  color: #555;\n  ", "\n  font-size: 0.9em;\n  font-weight: 600;\n"]);

  _templateObject = function _templateObject() {
    return data;
  };

  return data;
}

function _taggedTemplateLiteralLoose(strings, raw) { if (!raw) { raw = strings.slice(0); } strings.raw = raw; return strings; }

var StyledLabel = _styledComponents["default"].label(_templateObject(), _styledSystem.color);

var Label = function Label(_ref) {
  var children = _ref.children,
      required = _ref.required,
      rest = _objectWithoutPropertiesLoose(_ref, ["children", "required"]);

  return /*#__PURE__*/_react["default"].createElement(StyledLabel, _extends({}, rest, {
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 14,
      columnNumber: 5
    }
  }), required && /*#__PURE__*/_react["default"].createElement("abbr", {
    title: 'Campo obrigat√≥rio',
    __self: _this,
    __source: {
      fileName: _jsxFileName,
      lineNumber: 15,
      columnNumber: 20
    }
  }, "*"), ' ' + children);
};

var _default = Label;
exports["default"] = _default;