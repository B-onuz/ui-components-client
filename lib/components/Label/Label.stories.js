'use strict'

exports.__esModule = true
exports.LabelExample = exports['default'] = void 0

var _react = _interopRequireDefault(require('react'))

var _Label = _interopRequireDefault(require('./Label'))

var _this = void 0,
  _jsxFileName = '/home/junior/Fontes/ui-components-client/src/components/Label/Label.stories.js'

function _interopRequireDefault(obj) {
  return obj && obj.__esModule ? obj : { default: obj }
}

var _default = {
  title: 'Label',
  component: _Label['default'],
}
exports['default'] = _default

var LabelExample = function LabelExample() {
  return /*#__PURE__*/ _react['default'].createElement(
    _Label['default'],
    {
      __self: _this,
      __source: {
        fileName: _jsxFileName,
        lineNumber: 9,
        columnNumber: 35,
      },
    },
    'Label'
  )
}

exports.LabelExample = LabelExample
