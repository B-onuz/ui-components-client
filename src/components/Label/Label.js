import React from 'react'
import styled from 'styled-components'
import { color } from 'styled-system'

const StyledLabel = styled.label`
  color: #555;
  ${color}
  font-size: 0.9em;
  font-weight: 600;
`

const Label = ({ children, required, ...rest }) => {
  return (
    <StyledLabel {...rest}>
      {required && <abbr title={'Campo obrigatÃ³rio'}>*</abbr>}
      {' ' + children}
    </StyledLabel>
  )
}

export default Label
