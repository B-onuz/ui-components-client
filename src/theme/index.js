const colors = {
  red: '#db2828',
  lightRed: '#ef4d5a',
  mediumRed: '#FF4F5D',
  darkRed: '#9c2543',
  fullDarkRed: '#550e33',
  orange: '#f2711c',
  yellow: '#fbbd08',
  olive: '#b5cc18',
  green: '#21BA45',
  teal: '#00b5ad',
  blue: '#2185d0',
  facebookBlue: '#3B5998',
  blueLink: '#4285F4',
  violet: '#6435c9',
  purple: '#a333c8',
  pink: '#e03997',
  brown: '#a5673f',
  grey: '#767676',
  textGrey: '#8D8D8D',
  lightGrey: '#6D6D6D',
  mediumGrey: '#838383',
  darkGrey: '#9A9A9A',
  boldGray: '#55555A',
  black: '#1b1c1d',
  white: '#fdfdfd',
  warning: '#7A4D05',
  success: '#1E561F',
  info: '#0E566C',
  danger: '#973937',
  cyan: '#1CB5AD',
  disabledGrey: '#ffffff1f',
  disabledGreyText: '#ffffff4d',
}

colors.primary = colors.lightRed
colors.secondary = colors.fullDarkRed
colors.default = '#e0e1e2'

const breakpoints = ['40em', '52em', '64em', '80em']

breakpoints.sm = breakpoints[0]
breakpoints.md = breakpoints[1]
breakpoints.lg = breakpoints[2]
breakpoints.xl = breakpoints[3]

const fontSizes = [12, 14, 16, 20, 24, 32]

export default {
  fontFamily: `'Lato', sans-serif`,
  colors,
  fontSizes,
  breakpoints,
}
