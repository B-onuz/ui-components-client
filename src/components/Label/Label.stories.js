import React from 'react'
import Label from './Label'

export default {
  title: 'Label',
  component: Label,
}

export const LabelExample = () => <Label>Label</Label>
