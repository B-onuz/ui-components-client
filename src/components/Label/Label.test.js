import React from 'react'
import Label from './Label.js'
import renderer from 'react-test-renderer'
import { ThemeProvider } from 'styled-components'
import theme from '../../theme'
import 'jest-styled-components'

test('Label changes the class when hovered', () => {
  const component = renderer.create(
    <ThemeProvider theme={theme}>
      <Label>Hello Label</Label>
    </ThemeProvider>
  )
  let tree = component.toJSON()
  expect(tree).toMatchSnapshot()
})
