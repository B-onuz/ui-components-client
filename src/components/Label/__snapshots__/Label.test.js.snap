// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Label changes the class when hovered 1`] = `
.c0 {
  color: #555;
  font-size: 0.9em;
  font-weight: 600;
}

<label
  className="c0"
>
   Hello Label
</label>
`;
